server/src/main/java/org/elasticsearch/index/engine/LiveVersionMap.java

```diff
@@ -313,8 +344,8 @@ void maybePutUnderLock(BytesRef uid, VersionValue version) {
             putUnderLock(uid, version, maps);
         } else {
             maps.current.markAsUnsafe();
-            assert putAssertionMap(uid, version);
         }
+        assert putAssertionMap(uid, version);
```

Comment:
I like that this has moved, but I think it changes things a little. The thing that's currently called `unsafeKeysMap` is now a proper copy of the version map that's always safe, so deserves renaming (and the comment above its field declaration needs fixing). Also I suspect `getVersionForAssert` could be changed to assert that the value in `unsafeKeysMap` is the value in the main map, when non-null.